src/main/java/com/uber/profiling/profilers/ThreadInfoProfiler.java

```diff
@@ -0,0 +1,100 @@
+package com.uber.profiling.profilers;
+
+import com.uber.profiling.Profiler;
+import com.uber.profiling.Reporter;
+import com.uber.profiling.util.AgentLogger;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * ThreadInfoProfiler is used to Collects the Thread Related Metrics.
+ */
+public class ThreadInfoProfiler extends ProfilerBase implements Profiler {
+    public final static String PROFILER_NAME = "ThreadInfo";
+    public final static AgentLogger logger = AgentLogger.getLogger(ThreadInfoProfiler.class.getName());
+    private long intervalMillis = Constants.DEFAULT_METRIC_INTERVAL;
+
+    private ThreadMXBean threadMXBean;
+    private long previousTotalThreadCount = 0L; // to keep track of Total Thread.
+
+    private Reporter reporter;
+
+    public ThreadInfoProfiler(Reporter reporter) {
+        setReporter(reporter);
+        init();
+    }
+
+    private void init() {
+        try {
+            this.threadMXBean = ManagementFactory.getThreadMXBean();
+        }
+        catch (Throwable ex) {
+            logger.warn("Failed to get Thread MXBean", ex);
+        }
+
+    }
+
+    public void setIntervalMillis(long intervalMillis) {
+        this.intervalMillis = intervalMillis;
+    }
+
+    @Override
+    public long getIntervalMillis() {
+        return intervalMillis;
+    }
+
+    @Override
+    public void setReporter(Reporter reporter) {
+        this.reporter = reporter;
+    }
+
+    @Override
+    public void profile() {
+
+        long totalThreadCount = 0L; // total Thread created so far since JVm Launch.
+        int liveThreadCount = 0; // Number of thread which are currently active.
+        int peakThreadCount = 0; // the peak live thread count since the Java virtual machine started or peak was reset
+        long newThreadCount = 0; // Number of new thread created since last time time the metrics was created.
+        // This is a Derived metrics from previous data point.
+        if (threadMXBean != null) {
+            liveThreadCount =  threadMXBean.getThreadCount();
+            peakThreadCount = threadMXBean.getPeakThreadCount();
+            totalThreadCount = threadMXBean.getTotalStartedThreadCount();
```

Comment:
could we rename totalThreadCount to totalStartedThreadCount? totalStartedThreadCount matches the thread mxbean method name, thus is more clear for its meaning.

Reply:
sure.. ll update it.